contracts/WhitelistedCommitment.sol

```diff
@@ -89,35 +87,42 @@ contract WhitelistedCommitment is AccessControlled, AccessRoles, PublicCommitmen
             curve.burnNeumark(neumarks);
         }
     }
-    event Debug(uint256 value);
-    /// overrides base class to compute neumark reward for fixed price investors
+
+    /// overrides base class to compute neumark reward for ordered whitelistfix investors
     function giveNeumarks(address investor, uint256 eth, uint256 euros)
         internal
         returns (uint256)
     {
-        uint256 fixedTicket = fixedCost[investor]; // returns 0 in case of investor not in mapping
-        Debug(fixedTicket);
-        Debug(eth);
+        uint256 fixedRemainingTicket = fixedCostTickets[investor]; // returns 0 in case of investor not in mapping
+        uint256 fixedRemainingNeumarks = fixedCostNeumarks[investor]; // returns 0 in case of investor not in mapping
+
         // what is above limit for fixed price should be rewarded from curve
         uint256 reward = 0;
-        if ( eth > fixedTicket ) {
-            if (fixedTicket > 0) // recompute euro if part of msg.value goes thru whitelist
-                euros = convertToEUR(eth - fixedTicket);
-            reward = curve.issue(euros); // PublicCommitment.giveNeumarks(investor, eth - fixedTicket, euros);
-            eth = fixedTicket;
+        if ( eth > fixedRemainingTicket ) {
+            if (fixedRemainingTicket > 0) // recompute euro if part of msg.value goes thru whitelist
+                euros = convertToEUR(eth - fixedRemainingTicket);
+            reward = curve.issue(euros);
+            eth = fixedRemainingTicket;
         }
-        Debug(eth);
+
         // get pro rata neumark reward for any eth left
         uint256 fixedreward = 0;
         if (eth > 0) {
-            fixedreward = proportion(totalFixedCostNeumarks, eth, totalFixedCostAmount);
-            // rounding errors, send out remainders
-            // @remco review
-            uint256 remainingBalance = neumarkToken.balanceOf(address(this));
-            if (absDiff(fixedreward, remainingBalance) < 1000)
-                fixedreward = remainingBalance; // send all
-            // decrease ticket size
-            fixedCost[investor] -= eth;
+            fixedreward = proportion(fixedRemainingNeumarks, eth, fixedRemainingTicket);
+            // decrease ticket size and neumarks left
+            if ((fixedreward > fixedRemainingNeumarks)
```

Comment:
Please simplify the conditionals, it's hard to tell when something is executed and why. It should be super clear for critical code like this. Feel free to introduce boolean variables to explain things:

```
bool rewardOverflow = fixedreward > fixedRemainingNeumarks;
bool remainderTiny = fixedRemainingNeumarks - fixedreward < 10;
if (rewardOverflow || remainderTiny) {
    …
```